From b41ea0851b480017a44ec6e82f7257e9a3325548 Mon Sep 17 00:00:00 2001
From: Julien Bonjean <julien@bonjean.info>
Date: Sat, 30 Jan 2016 19:38:50 -0500
Subject: [PATCH 3/3] consume ovs monitor events

The events from the monitor were not consumed so has_updates always
returned true leading to many ovs operations on every run.
---
 .../neutron/plugins/ml2/drivers/openvswitch/agent/ovs_neutron_agent.py  | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/usr/lib/python2.7/dist-packages/neutron/plugins/ml2/drivers/openvswitch/agent/ovs_neutron_agent.py b/usr/lib/python2.7/dist-packages/neutron/plugins/ml2/drivers/openvswitch/agent/ovs_neutron_agent.py
index b18716e..1bbd545 100644
--- a/usr/lib/python2.7/dist-packages/neutron/plugins/ml2/drivers/openvswitch/agent/ovs_neutron_agent.py
+++ b/usr/lib/python2.7/dist-packages/neutron/plugins/ml2/drivers/openvswitch/agent/ovs_neutron_agent.py
@@ -1705,6 +1705,8 @@ class OVSNeutronAgent(sg_rpc.SecurityGroupAgentRpcCallbackMixin,
                     tunnel_sync = True
             ovs_restarted |= (ovs_status == constants.OVS_RESTARTED)
             if self._agent_has_updates(polling_manager) or ovs_restarted:
+                # consume events
+                polling_manager.get_events()
                 try:
                     LOG.debug("Agent rpc_loop - iteration:%(iter_num)d - "
                               "starting polling. Elapsed:%(elapsed).3f",
--
2.7.0.rc3
